docs/sts/tls.md

```diff
@@ -0,0 +1,126 @@
+# AssumeRoleWithCertificate [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io)
+
+## Introduction
+
+MinIO provides a custom STS API that allows authentication with client X.509 / TLS certificates.
+
+A major advantage of certificate-based authentication compared to other STS authentication methods,
+like OpenID Connect or LDAP/AD, is that client authentication works without any additional/external
+component that must be constantly available. Therefore, certificate-based authentication may provide
+better availability / lower operational complexity.
+
+The MinIO TLS STS API can be configured via MinIO's standard configuration API (i.e. using `mc admin config set/get`).
+Further, it can be configured via the following environment variables:
+
+```
+mc admin config set myminio identity_tls --env
+KEY:
+identity_tls  enable X.509 TLS certificate SSO support
+
+ARGS:
+MINIO_IDENTITY_TLS_STS_EXPIRY   (duration)  temporary credentials validity duration in s,m,h,d. Default is "1h"
+MINIO_IDENTITY_TLS_SKIP_VERIFY  (on|off)    trust client certificates without verification. Defaults to "off" (verify)
```

Comment:
Do we need a skip verify for TLS client auth? The purpose here is to authenticate the client, not just have a secure transport right?

Reply:
Yes, both. However, e.g. in case of LDAP an insecure transport is completely broken in the same way w.r.t. authentication. If you send your LDAP credentials over a plaintext connection / resp. a TLS connection without verifying to whom your talking then it's more or less trivial to obtain LDAP credentials. Either via inspecting the network traffic or via MitM.

The idea here is: You can use your self-signed certificates - e.g. to test that it works. Otherwise, you cannot test your client application without an actual certificate from your CA.